src/trade_executor.h

```diff
@@ -90,14 +112,36 @@ template <class T> class TradeExecutor {
         Trade<T>& trade = v_it->first;
         typename T::Ptr rsrc = v_it->second;
         if (rsrc->quantity() > cyclus::eps_rsrc()) {
+          // Calculate the adjusted preference that was actually used by the
+          // solver
+          double adjusted_preference = trade.bid->preference();
+
+          // If the bid has NaN preference, use the request preference
+          if (std::isnan(adjusted_preference)) {
+            adjusted_preference = trade.request->preference();
+          }
+
+          // If we have access to the exchange context, use the adjusted
+          // preference that was actually used by the solver
+          if (ex_ctx != NULL) {
```

Comment:
I think this is the same thing ...
```suggestion
          if (ex_ctx) {
```